### ISSUE TITLE
Forward pass: on 1 vector vs on a batch of vectors

### ISSUE THREAD

**Issue by vasavig** [soumith/dcgan.torch#48]
Hi,

After training the network with Celeb faces dataset, a forward pass on 10 noise vectors into the generator gives decent results like below:
![output_1](https://cloud.githubusercontent.com/assets/4658916/19799824/be19d5de-9cc6-11e6-8338-3d2ecb900e5a.png)
But when I passed same vector replicated as a batch of 10 to the trained generator it gave the following:
![output_2_same_normal](https://cloud.githubusercontent.com/assets/4658916/19799933/4a7b071e-9cc7-11e6-9668-340de6e0d7d0.png)
It looks like low frequencies are not present. Similar results were obtained when a single vector was passed.
I think I am missing something. Is the forward pass on trained generator affected by the number and kind of Z vectors passed?

Any insight is appreciated. Thanks!

**Comment by soumith** [comment]
there are batchnorm interactions that will be present.

If you want to avoid such interactions between batch samples, put the model in evaluate mode, i.e. G:evaluate() and then push the replicated noise vectors through.

**Comment by vasavig** [comment]
Thanks, soumith! That clears a lot of confusion.
